### ISSUE TITLE
Crashes following certain operations -[GCReference compareWithReference:]

### ISSUE THREAD

**Issue by dougsonos** [git-up/GitUp#185]
Often Gitup crashes on macOS 10.12 following certain operations that rewrite commits. I don't have a reproducible pattern yet. Attached is a crash log.

```
Crashed Thread:        0  Dispatch queue: com.apple.main-thread

Exception Type:        EXC_BAD_ACCESS (SIGSEGV)
Exception Codes:       KERN_INVALID_ADDRESS at 0x0000000000000050
Exception Note:        EXC_CORPSE_NOTIFY

VM Regions Near 0x50:
--> 
    __TEXT                 0000000101b05000-0000000101b6b000 [  408K] r-x/rwx SM=COW  /Users/USER/Downloads/GitUp.app/Contents/MacOS/GitUp

Application Specific Information:
Performing @selector(undo:) from sender NSMenuItem 0x6000002a1380

Thread 0 Crashed:: Dispatch queue: com.apple.main-thread
0   libsystem_kernel.dylib        	0x00007fff8f500dda __pthread_kill + 10
1   libsystem_pthread.dylib       	0x00007fff8f5eb797 pthread_kill + 90
2   libsystem_c.dylib             	0x00007fff8f4174d7 raise + 26
3   co.gitup.mac                  	0x0000000101b2d88a 0x101b05000 + 166026
4   libsystem_platform.dylib      	0x00007fff8f5ded7a _sigtramp + 26
5   ???                           	000000000000000000 0 + 0
6   co.gitup.kit                  	0x0000000101c13683 -[GCReference compareWithReference:] + 48
7   co.gitup.kit                  	0x0000000101c137a0 -[GCReference(Extensions) isEqualToReference:] + 35
8   co.gitup.mac                  	0x0000000101b1778f 0x101b05000 + 75663
9   co.gitup.kit                  	0x0000000101c7b5f6 -[GIMapViewController _reloadMap:] + 764
10  com.apple.CoreFoundation      	0x00007fff7a32f5dc __CFNOTIFICATIONCENTER_IS_CALLING_OUT_TO_AN_OBSERVER__ + 12
11  com.apple.CoreFoundation      	0x00007fff7a32f4db _CFXRegistrationPost + 427
12  com.apple.CoreFoundation      	0x00007fff7a32f242 ___CFXNotificationPost_block_invoke + 50
13  com.apple.CoreFoundation      	0x00007fff7a2ed962 -[_CFXNotificationRegistrar find:object:observer:enumerator:] + 2018
14  com.apple.CoreFoundation      	0x00007fff7a2ec94b _CFXNotificationPost + 667
15  com.apple.Foundation          	0x00007fff7bd2b2a3 -[NSNotificationCenter postNotificationName:object:userInfo:] + 66
16  co.gitup.kit                  	0x0000000101c0da2e -[GCLiveRepository _updateHistory] + 502
17  co.gitup.kit                  	0x0000000101c0d4ad -[GCLiveRepository _notifyWorkingDirectoryChanged:gitDirectoryChanged:] + 329
18  co.gitup.kit                  	0x0000000101c0fc44 -[GCLiveRepository _undoOperationWithReason:beforeSnapshot:afterSnapshot:checkoutIfNeeded:ignore:] + 999
19  com.apple.CoreFoundation      	0x00007fff7a31732c __invoking___ + 140
20  com.apple.CoreFoundation      	0x00007fff7a3171b1 -[NSInvocation invoke] + 289
21  com.apple.CoreFoundation      	0x00007fff7a330316 -[NSInvocation invokeWithTarget:] + 54
22  com.apple.Foundation          	0x00007fff7be020e5 -[_NSUndoStack popAndInvoke] + 235
23  com.apple.Foundation          	0x00007fff7be01e8c -[NSUndoManager undoNestedGroup] + 433
24  libsystem_trace.dylib         	0x00007fff8f60419d _os_activity_initiate + 61
25  com.apple.AppKit              	0x00007fff786c769f -[NSApplication(NSResponder) sendAction:to:from:] + 456
26  com.apple.AppKit              	0x00007fff782070ce -[NSMenuItem _corePerformAction] + 324
27  com.apple.AppKit              	0x00007fff78206e3a -[NSCarbonMenuImpl performActionWithHighlightingForItemAtIndex:] + 114
28  libsystem_trace.dylib         	0x00007fff8f60419d _os_activity_initiate + 61
29  com.apple.AppKit              	0x00007fff78205c4e -[NSMenu performKeyEquivalent:] + 367
30  com.apple.AppKit              	0x00007fff786c6281 routeKeyEquivalent + 1024
31  com.apple.AppKit              	0x00007fff786c4c1d -[NSApplication(NSEvent) sendEvent:] + 3149
32  com.apple.AppKit              	0x00007fff77faae8c -[NSApplication run] + 1002
33  com.apple.AppKit              	0x00007fff77f75976 NSApplicationMain + 1237
34  libdyld.dylib                 	0x00007fff8f3d2255 start + 1
```

[GitUp_2016-08-02-085821.crash.txt](https://github.com/git-up/GitUp/files/397606/GitUp_2016-08-02-085821.crash.txt)

**Comment by AndrewHoos** [comment]
This looks like the same backtrace of a crash I have seen easiest reproduce method I have seen in 10.12:

1. Checkout master
2. Double click on master^ (one commit off of tip of branch)
3. Double click on master (current commit)

**Comment by swisspol** [comment]
Doesn't crash for me in a random repo I just tested with, so there must be some other pre-conditions.

**Comment by swisspol** [comment]
Looks like a dup of #185.

**Comment by swisspol** [comment]
If you can repro the crash using some open source repo that be checked out, please let me know!

**Comment by swisspol** [comment]
According to Crashlytics, there were 581 crashes affecting 107 users in the past 30 days due to this very bug. This is by very very far the #1 GitUp crash.

What's interesting though is that CrashLytics reports that all the crashes happened on 10.12.

**Comment by AndrewHoos** [comment]
I was able to reproduce this with the gitup repo itself
1. cloning opening with gitup command line too
2. double clicking origin/master^ 
3. double clicking origin/master

Is this potentialy related to new version of git/devtools in 10.12/Xcode beta?

**Comment by swisspol** [comment]
Are you reproducing this on 10.12? Is that from running GitUp from Xcode or using the official build? Can you paste in a full stack trace (with symbols)? Thanks.

**Comment by AndrewHoos** [comment]
I am running the release (stable) version of gitup
[GitUp_2016-08-31-151027_hoos-tungsten-2.crash.txt](https://github.com/git-up/GitUp/files/448372/GitUp_2016-08-31-151027_hoos-tungsten-2.crash.txt)

**Comment by swisspol** [comment]
This should be fixed by 16d3848151b6856ae724706083282efa28594f7e which is in build 1019. Let me know ASAP if this crash still happens.

**Comment by AndrewHoos** [comment]
Confirmed this is fixed by 16d3848
